FirebaseRemoteConfig/Sources/FIRRemoteConfig.m

```diff
@@ -331,6 +358,10 @@ - (FIRRemoteConfigValue *)configValueForKey:(NSString *)key {
                     @"Key %@ should come from source:%zd instead coming from source: %zd.", key,
                     (long)FIRRemoteConfigSourceRemote, (long)value.source);
       }
+      if ([self->_configContent.activePersonalization count] > 0) {
+        [self callListeners:value.stringValue
+                   metadata:self->_configContent.activePersonalization[key]];
```

Comment:
thinking: we mentioned that the listener approach would allow us to add other side-effects in the future, but given the current implementation all the listeners would be called specifically with personalization metadata, which limits the functionality we can add